qa/setup_packages.py

```diff
@@ -16,34 +16,433 @@
+                Cuda version used for this query. Disregarded
+        """
+        return self.name
+
+    def get_all_versions(self, cuda_version=None):
+        """Get all versions compatible with provided cuda_version
+
+            Parameters
+            ----------
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return self.filter_versions(self.versions)
+
+    def get_install_string(self, idx, cuda_version=None):
+        """Obtains installation string that pip should accept for version at
+            a given index with a given cuda_version
+
+            Parameters
+            ----------
+            `idx`: int
+                Index of version to retrive. If index is beyond 0-num_ov_versions-1 range
+                it is clamped to it
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return "{name}=={version}".format(name=self.get_name(cuda_version), version=self.get_version(idx, cuda_version))
+
+    def get_version_string(self, idx, cuda_version=None):
+        """Obtains version string for version at a given index
+            with a given cuda_version
+
+            Parameters
+            ----------
+            `idx`: int
+                Index of version to retrive. If index is beyond 0-num_ov_versions-1 range
+                it is clamped to it
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return "{version}".format(version=self.get_version(idx, cuda_version))
+
+class CudaPackage(PlainPackage):
+    """Class describing a cuda package with a key/name and a dictionary where the key
+        is a cuda version and value is the list of versions supported.
+
+        Parameters
+        ----------
+        `key`: str
+            Name this package should be queried for
+        `versions`: str
+            Dictionary, where the key is a cuda version and vale, is the list of versions supported
+        `name`: str, , optional, default = None
+            Name of this package used during installation. If empty it is the same as the key.
+            If it includes `{cuda_v}` it is replaced by the cuda_version when queried
+    """
+    def __init__(self, key, versions, name=None):
+        super(CudaPackage, self).__init__(key, versions, name)
+        if not isinstance(versions, dict):
+            raise TypeError("versions argument should by dict type [cuda_version : list_of_versions")
+
+    def get_name(self, cuda_version):
+        """Retrives package name. It formats package name using cuda version
+            as a replacement for `{cuda_v}` in the package name
+
+            Parameters
+            ----------
+            `cuda_version`: str
+                Cuda version used for this query
+        """
```

Comment:
Just a thought, but I don't think it's that necessary to have the docstrings for all the overloaded methods of derived classes.

Reply:
It makes the file longer. The short description for class that's package names/install commands are cuda dependent or not would be probably enough.

Reply:
Done